app/src/main/java/org/openedx/app/di/AppModule.kt

```diff
@@ -153,4 +157,6 @@ val appModule = module {
     factory { FacebookAuthHelper() }
     factory { GoogleAuthHelper(get()) }
     factory { MicrosoftAuthHelper() }
+
+    single<EnrollInCourseInteractor> { get<CourseInteractor>() }
```

Comment:
Do we need to keep this dependency as a singleton?
CourseInteractor, by nature, does not keep any state or data, so it seems there is no reason to keep EnrollInCourseInteractor as singleton.

Reply:
updated